manim/mobject/geometry.py

```diff
@@ -1630,48 +1630,49 @@ def construct(self):
         :class:`RoundedRectangle`
         """
 
-        new_points = []
-
-        for vertices in self.get_vertex_groups():
-            arcs = []
-            for v1, v2, v3 in adjacent_n_tuples(vertices, 3):
-                vect1 = v2 - v1
-                vect2 = v3 - v2
-                unit_vect1 = normalize(vect1)
-                unit_vect2 = normalize(vect2)
-
-                angle = angle_between_vectors(vect1, vect2)
-                # Negative radius gives concave curves
-                angle *= np.sign(radius)
-
-                # Distance between vertex and start of the arc
-                cut_off_length = radius * np.tan(angle / 2)
-
-                # Determines counterclockwise vs. clockwise
-                sign = np.sign(np.cross(vect1, vect2)[2])
-
-                arc = ArcBetweenPoints(
-                    v2 - unit_vect1 * cut_off_length,
-                    v2 + unit_vect2 * cut_off_length,
-                    angle=sign * angle,
-                )
-                arcs.append(arc)
+        if radius is not None and radius is not 0:
```

Comment:
This should instead be
```py
if radius is None or radius == 0:
    return self

...
```
The success condition should always be the main trunk of the code since it improves readability a lot as you don't get a lot of nesting and you can right away see that nothing is done if `radius` is `None` or 0, whereas in the current code you have to wait to get to the bottom of the if condition to see that.

Additionally the typing should be adjusted to accept `None`, so `Optional[<whatever it is now>]`

Reply:
The reason that I left `return self` until the bottom is that it needs called either way, so this removes duplicating the return twice.

Regarding `None`, can you be more clear what you mean by "some appropriate radius for the shape"? My thinking is that if you want a radius _of None_, then you effectively want a radius of zero. Because I think they're the same idea in this case, I checked against both because users could reasonably pass either (and I know this isn't JavaScript, but also because JS would agree that they represent the same thing, hence the difference between `==` and `===`).

Reply:
> The reason that I left return self until the bottom is that it needs called either way, so this removes duplicating the return twice.

I'd really rather you duplicate the `return self` (for my stated reasons) than nest the meat of the function. It really, really improves readability and I'm sure we'd both agree that in an extreme case of a lot of different "failure" conditions, you wouldn't want nests on top of nests on top of nests just so you can only have one `return self`.

> Regarding `None,` can you be more clear what you mean by "some appropriate radius for the shape"?

What passing `radius=None` would signal to me is "I don't care what value the radius is, just round the corners", whereas passing `radius=0` would signal "I don't want any curvature for the corners".

Reply:
I still disagree about `None`, but because I mainly care about checking for `0`, I changed it.